### PR TITLE
test: 20% coverage of add-edit-per-diem

### PR DESCRIPTION
**Pull request by suyashpatil78**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=dwwESE3)
### <samp>🤖 Generated by Copilot at a0892f9</samp>

This pull request adds and updates some features and tests related to per diem expenses. It modifies the mock data and interface for org categories and default transaction field values to include per diem specific properties. It also improves the unit tests and code quality for the `add-edit-per-diem` page.

### <samp>🤖 Generated by Copilot at a0892f9</samp>

> _Mocking the data for the fields of doom_
> _Per diem expenses in the test room_
> _Cleaning the code for the org categories_
> _Unit testing the `add-edit-per-diem` page with ease_

### <samp>🤖 Generated by Copilot at a0892f9</samp>

*  Add mock data objects for per diem expenses and default transaction field values ([link](https://github.com/fylein/fyle-mobile-app/pull/2254/files?diff=unified&w=0#diff-88d03c9481ee9f872719d0aa13033ce30c0e50a9d5de1403d54b362a3f5282cbR14-R22), [link](https://github.com/fylein/fyle-mobile-app/pull/2254/files?diff=unified&w=0#diff-8914bfb32bd70efe52e673bc9938855b2c84a494d9464f82c50fe2c8bfbdca45R989-R1001))
*  Add properties to the default transaction field values interface for per diem expenses ([link](https://github.com/fylein/fyle-mobile-app/pull/2254/files?diff=unified&w=0#diff-affd857cff1386af542b2419395498ea27a204e1b6634747ab18b2509d0a5369R11-R15))
*  Remove unused import of `OrgCategoryListItem` from mock data file ([link](https://github.com/fylein/fyle-mobile-app/pull/2254/files?diff=unified&w=0#diff-8914bfb32bd70efe52e673bc9938855b2c84a494d9464f82c50fe2c8bfbdca45L1-R1))
*  Update imports in the spec file for the add-edit-per-diem page ([link](https://github.com/fylein/fyle-mobile-app/pull/2254/files?diff=unified&w=0#diff-7e625c752cfe2c440ad708a7d679ac2c9a02de955c032be750e76dd4532bf933L39-R53))
*  Add test cases for the methods of the add-edit-per-diem page using mock data and services ([link](https://github.com/fylein/fyle-mobile-app/pull/2254/files?diff=unified&w=0#diff-7e625c752cfe2c440ad708a7d679ac2c9a02de955c032be750e76dd4532bf933R490-R613))

## Clickup
https://app.clickup.com/t/85ztk93dv

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem-1.page.spec.ts [hunk 1]
```diff
@@ -477,5 +487,163 @@ export function TestCases1(getTestBed) {
...
+          expenseFieldsMapResponse,
+          fields,
+          perDiemCategory
+        );
+        expect(res).toEqual(mockTxnFieldData);
+        done();
+      });
+    });
+
+    it('setupTfcDefaultValues(): should update form with expense field value if some fields are empty', () => {
```
Comment (Dimple16) [inline src/app/fyle/add-edit-per-diem/add-edit-per-diem-1.page.spec.ts:515]: `should update the form with default expense field values if some fields are empty`

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem-1.page.spec.ts [hunk 2]
```diff
@@ -477,5 +487,163 @@ export function TestCases1(getTestBed) {
...
+      accountsService.getPaymentModes.and.returnValue(paymentModesData);
+      fixture.detectChanges();
+
+      component.getPaymentModes().subscribe((res) => {
+        expect(res).toEqual(paymentModesData);
+        expect(accountsService.getEMyAccounts).toHaveBeenCalledTimes(1);
+        expect(orgSettingsService.get).toHaveBeenCalledTimes(1);
+        expect(orgUserSettingsService.getAllowedPaymentModes).toHaveBeenCalledTimes(1);
+        expect(paymentModesService.checkIfPaymentModeConfigurationsIsEnabled).toHaveBeenCalledTimes(1);
+        const config = {
```
Comment (Dimple16) [inline src/app/fyle/add-edit-per-diem/add-edit-per-diem-1.page.spec.ts:565]: move this config to a separate mock data file

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem-1.page.spec.ts [hunk 3]
```diff
@@ -477,5 +487,163 @@ export function TestCases1(getTestBed) {
...
+        });
+      });
+
+      it('should return empty array if category id is undefined', (done) => {
+        const mockPerDiemCategory = cloneDeep(perDiemCategory);
+        mockPerDiemCategory.id = undefined;
+        categoriesService.getAll.and.returnValue(of([...expectedAllOrgCategories, mockPerDiemCategory]));
+        component.getProjectCategoryIds().subscribe((res) => {
+          expect(categoriesService.getAll).toHaveBeenCalledTimes(1);
+          expect(res).toEqual([undefined]);
```
Comment (Dimple16) [inline src/app/fyle/add-edit-per-diem/add-edit-per-diem-1.page.spec.ts:631]: the description says `return empty array`, why is `[undefined]` required?

**Review by Dimple16 — APPROVED** [review]
minor comments

**Review by Chethan-Fyle — APPROVED** [review]

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem-1.page.spec.ts (hunk 3 above)
Comment (OmkarJ13) [reply src/app/fyle/add-edit-per-diem/add-edit-per-diem-1.page.spec.ts:631]: ➕

**Review by OmkarJ13 — APPROVED** [review]

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem-1.page.spec.ts (hunk 3 above)
Comment (suyashpatil78) [reply src/app/fyle/add-edit-per-diem/add-edit-per-diem-1.page.spec.ts:628]: I have corrected the description and added comment as well.

**Pull request merged**